pallets/transaction-payment/src/lib.rs

```diff
@@ -392,21 +464,23 @@ where
         info: &DispatchInfoOf<T::Call>,
         len: usize,
     ) -> Result<(BalanceOf<T>, Option<NegativeImbalanceOf<T>>), TransactionValidityError> {
-        let fee = Module::<T>::compute_fee(len as u32, info, 0u32.into());
+        let tip = self.0;
+        let fee = Module::<T>::compute_fee(len as u32, info, tip);
 
         // Only mess with balances if fee is not zero.
         if fee.is_zero() {
             return Ok((fee, None));
         }
 
         if let Some(payer_key) = T::CddHandler::get_valid_payer(call, &who)? {
-            let imbalance = T::Currency::withdraw(
-                &payer_key,
-                fee,
-                WithdrawReason::TransactionPayment.into(),
-                ExistenceRequirement::KeepAlive,
-            )
-            .map_err(|_| InvalidTransaction::Payment)?;
+            let reason = if tip.is_zero() {
+                WithdrawReason::TransactionPayment.into()
+            } else {
+                WithdrawReason::TransactionPayment | WithdrawReason::Tip
```

Comment:
Since tip is always 0, withdraw reason is always `WithdrawReason::TransactionPayment.into()`

Reply:
Yep, already fixed :)